### PR TITLE
Fix for ModelAutocompleteType does not display the previously selected value when editing an entity

### LINKED ISSUE #7210: ModelAutocompleteType does not display the previously selected value when editing an entity
#### Sonata packages

<details><summary>show</summary>
<p>

```
sonata-project/admin-bundle              3.100.1 3.100.1 The missing Symfony Admin Generator
sonata-project/block-bundle              3.23.1  4.5.3   Symfony SonataBlockBundle
sonata-project/cache                     2.1.1   2.1.1   Cache library
sonata-project/core-bundle               3.20.0  3.20.0  Symfony SonataCoreBundle (abandoned)
sonata-project/doctrine-extensions       1.12.0  1.12.0  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.34.1  3.34.1  Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  2.6.2   2.6.2   Lightweight Exporter library
sonata-project/form-extensions           0.1.2   1.9.0   Symfony form extensions
sonata-project/twig-extensions           0.1.1   1.6.0   Sonata twig extensions
```

</p>
</details>

#### Symfony packages

<details><summary>show</summary>
<p>

```
symfony/asset                      v4.4.22 v5.2.7  Manages URL generation and versioning of web assets such as CSS stylesheets, JavaScript files and image files
symfony/browser-kit                v4.4.24 v5.2.9  Simulates the behavior of a web browser, allowing you to make requests, click on links and submit forms programmatically
symfony/cache                      v4.4.24 v5.2.9  Provides an extended PSR-6, PSR-16 (and tags) implementation
symfony/cache-contracts            v2.4.0  v2.4.0  Generic abstractions related to caching
symfony/config                     v4.4.23 v5.2.8  Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/console                    v4.4.24 v5.2.8  Eases the creation of beautiful and testable command line interfaces
symfony/css-selector               v4.4.24 v5.2.9  Converts CSS selectors to XPath expressions
symfony/debug                      v4.4.22 v4.4.22 Provides tools to ease debugging PHP code
symfony/debug-bundle               v4.4.20 v5.2.4  Provides a tight integration of the Symfony Debug component into the Symfony full-stack framework
symfony/dependency-injection       v4.4.24 v5.2.9  Allows you to standardize and centralize the way objects are constructed in your application
symfony/deprecation-contracts      v2.4.0  v2.4.0  A generic function and convention to trigger deprecation notices
symfony/doctrine-bridge            v4.4.24 v5.2.9  Provides integration for Doctrine with various Symfony components
symfony/dom-crawler                v4.4.24 v5.2.9  Eases DOM navigation for HTML and XML documents
symfony/dotenv                     v4.4.20 v5.2.4  Registers environment variables from a .env file
symfony/error-handler              v4.4.23 v5.2.8  Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher           v4.4.20 v5.2.4  Provides tools that allow your application components to communicate with each other by dispatching events and listening to them
symfony/event-dispatcher-contracts v1.1.9  v2.4.0  Generic abstractions related to dispatching event
symfony/expression-language        v4.4.22 v5.2.7  Provides an engine that can compile and evaluate expressions
symfony/filesystem                 v4.4.22 v5.2.7  Provides basic utilities for the filesystem
symfony/finder                     v4.4.24 v5.2.9  Finds files and directories via an intuitive fluent interface
symfony/flex                       v1.13.3 v1.13.3 Composer plugin for Symfony
symfony/form                       v4.4.24 v5.2.9  Allows to easily create, process and reuse HTML forms
symfony/framework-bundle           v4.4.24 v5.2.9  Provides a tight integration between Symfony components and the Symfony full-stack framework
symfony/http-client                v4.4.24 v5.2.9  Provides powerful methods to fetch HTTP resources synchronously or asynchronously
symfony/http-client-contracts      v2.4.0  v2.4.0  Generic abstractions related to HTTP clients
symfony/http-foundation            v4.4.23 v5.2.8  Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                v4.4.24 v5.2.9  Provides a structured process for converting a Request into a Response
symfony/inflector                  v4.4.23 v5.2.8  Converts words between their singular and plural forms (English only)
symfony/intl                       v4.4.22 v5.2.7  Provides a PHP replacement layer for the C intl extension that includes additional data from the ICU library
symfony/mailer                     v4.4.22 v5.2.7  Helps sending emails
symfony/maker-bundle               v1.31.1 v1.31.1 Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you can forget about writing boilerplate code.
symfony/mime                       v4.4.24 v5.2.9  Allows manipulating MIME messages
symfony/monolog-bridge             v4.4.22 v5.2.7  Provides integration for Monolog with various Symfony components
symfony/monolog-bundle             v3.7.0  v3.7.0  Symfony MonologBundle
symfony/options-resolver           v4.4.20 v5.2.4  Provides an improved replacement for the array_replace PHP function
symfony/phpunit-bridge             v5.2.9  v5.2.9  Provides utilities for PHPUnit, especially user deprecation notices management
symfony/polyfill-intl-grapheme     v1.22.1 v1.22.1 Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu          v1.22.1 v1.22.1 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.22.1 v1.22.1 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer   v1.22.1 v1.22.1 Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring          v1.22.1 v1.22.1 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.22.1 v1.22.1 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.22.1 v1.22.1 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-php80             v1.22.1 v1.22.1 Symfony polyfill backporting some PHP 8.0+ features to lower PHP versions
symfony/process                    v4.4.22 v5.2.7  Executes commands in sub-processes
symfony/property-access            v4.4.20 v5.2.4  Provides functions to read and write from/to an object or array using a simple string notation
symfony/property-info              v4.4.23 v5.2.8  Extracts information about PHP class' properties using metadata of popular sources
symfony/routing                    v4.4.24 v5.2.9  Maps an HTTP request to a set of configuration variables
symfony/security-acl               v3.1.2  v3.1.2  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.4.23 v5.2.9  Provides a tight integration of the Security component into the Symfony full-stack framework
symfony/security-core              v4.4.24 v5.2.9  Symfony Security Component - Core Library
symfony/security-csrf              v4.4.22 v5.2.7  Symfony Security Component - CSRF Library
symfony/security-guard             v4.4.23 v5.2.8  Symfony Security Component - Guard
symfony/security-http              v4.4.24 v5.2.9  Symfony Security Component - HTTP Integration
symfony/serializer                 v4.4.24 v5.2.9  Handles serializing and deserializing data structures, including object graphs, into array structures or other formats like XML and JSON.
symfony/service-contracts          v2.4.0  v2.4.0  Generic abstractions related to writing services
symfony/stopwatch                  v4.4.20 v5.2.7  Provides a way to profile code
symfony/string                     v5.2.8  v5.2.8  Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and grapheme clusters in a unified way
symfony/templating                 v4.4.20 v5.2.4  Provides all the tools needed to build any kind of template system
symfony/translation                v4.4.24 v5.2.9  Provides tools to internationalize your application
symfony/translation-contracts      v2.4.0  v2.4.0  Generic abstractions related to translation
symfony/twig-bridge                v4.4.22 v5.2.8  Provides integration for Twig with various Symfony components
symfony/twig-bundle                v4.4.24 v5.2.9  Provides a tight integration of Twig into the Symfony full-stack framework
symfony/validator                  v4.4.24 v5.2.9  Provides tools to validate values
symfony/var-dumper                 v4.4.22 v5.2.8  Provides mechanisms for walking through any arbitrary PHP variable
symfony/var-exporter               v4.4.23 v5.2.8  Allows exporting any serializable PHP data structure to plain PHP code
symfony/web-link                   v4.4.21 v5.2.5  Manages links between resources
symfony/web-profiler-bundle        v4.4.23 v5.2.8  Provides a development tool that gives detailed information about the execution of any request
symfony/webpack-encore-bundle      v1.11.2 v1.11.2 Integration with your Symfony app & Webpack Encore!
symfony/yaml                       v4.4.24 v5.2.9  Loads and dumps YAML files
```

</p>
</details>

#### PHP version

```
PHP 7.3.28 (cli) (built: May 12 2021 13:40:34) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.28, Copyright (c) 1998-2018 Zend Technologies
```

## Subject

When using ModelAutocompleteType to allow the selection of a related entity the functionality to select the entity and then save the form persisting the data works perfectly. However, if you then return to the form to edit the entity the value of the selected related entity is not displayed.

## Minimal repository with the bug

```
protected function configureFormFields(FormMapper $formMapper): void
    {
        $subject = $formMapper->getAdmin()->getSubject();

        $formMapper
            ->add(
                'merchant',
                ModelAutocompleteType::class,
                [
                    'property' => 'name',
                    'minimum_input_length' => 2,
                    'required'   => true,
                    'btn_add' => 'Add new merchant'
                ]
            );
    }

```

## Expected results

I would expect the '__toString()' representation of the selected related entity to be visible in the input. 

## Actual results

The input is blank. The value is present in a hidden input so the entity can be saved and the value remains unchanged but it is not possible to see the value within the form.

### PR DESCRIPTION
**Pull request by willemverspyck**
I am targeting this branch, because it's BC.

Closes #7210 

## Changelog

```markdown

### Changed
Reverted the changes of #7196, because they only work for Select2 version 4, that is used on the Master branche. Branche 3.x uses Select2 version 3.

```

### CODE REVIEW

**Review by VincentLanglet — DISMISSED** [review]

**Comment by VincentLanglet** [comment]
Maybe we should add a comment `NEXT_MAJOR` in order to keep the `attr` method in master (when doing the merge 3.x into master).

**Comment by willemverspyck** [comment]
@VincentLanglet I've added the NEXT_MAJOR comment. Hope this is the correct way.

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig [hunk 1]
```diff
@@ -308,10 +308,12 @@ file that was distributed with this source code.
                         {% endif %}
 
                         // append to Select2
-                        autocompleteInput.attr('data', data).append(data).trigger('change');
+                        // NEXT_MAJOR: Remove ".select2('data', data)" while dropping v3 support
+                        autocompleteInput.select2('data', data).append(data).trigger('change');
 
                         // manually trigger the `select2:select` event
-                        autocompleteInput.attr('data', data).trigger({
+                        // NEXT_MAJOR: Remove ".select2('data', data)" while dropping v3 support
```
Comment (VincentLanglet) [inline src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:315]:
```suggestion
                        // NEXT_MAJOR: Change to ".attr('data', data)" while dropping v3 support
```

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig [hunk 2]
```diff
@@ -308,10 +308,12 @@ file that was distributed with this source code.
                         {% endif %}
 
                         // append to Select2
-                        autocompleteInput.attr('data', data).append(data).trigger('change');
+                        // NEXT_MAJOR: Remove ".select2('data', data)" while dropping v3 support
```
Comment (VincentLanglet) [inline src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:311]:
```suggestion
                        // NEXT_MAJOR: Change to ".attr('data', data)" while dropping v3 support
```

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig (hunk 1 above)
Comment (willemverspyck) [reply src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:315]:
```
Can you try this ? @alistairstead @willemverspyck

// append to Select2
autocompleteInput.append(data).trigger('change');

// manually trigger the `select2:select` event
autocompleteInput.trigger({
    type: 'select2:select',
    params: {
        data: data
    }
});
This changes works for me too (master) and its better code. @VincentLanglet
```

As mentioned by @cecicifu in #7210, we can remove ".select2('data', data)". I also tested this in the master branch and that works. Same for line 311.

Comment (VincentLanglet) [reply src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:315]: Ok thanks

**Review by VincentLanglet — APPROVED** [review]

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig (hunk 1 above)
Comment (cecicifu) [reply src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:315]: The improvement i have made is only for master branch, need to open a new PR in master branch, right ?

Comment (VincentLanglet) [reply src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:315]: I will ping you when I'll do the merge of 3.x into master after this merge @cecicifu

**Pull request merged**